opencti-platform/opencti-graphql/src/modules/publicDashboard/publicDashboard-domain.ts

```diff
@@ -247,6 +275,8 @@ export const publicDashboardDelete = async (context: AuthContext, user: AuthUser
 // heatmap & vertical-bar & line & area
 export const publicStixCoreObjectsMultiTimeSeries = async (context: AuthContext, args: QueryPublicStixCoreObjectsMultiTimeSeriesArgs) => {
   const { user, parameters, dataSelection } = await getWidgetArguments(context, args.uriKey, args.widgetId);
+  context.user = user;
```

Comment:
why do we need to put the user in the context ?

Reply:
As the request is made by a non connected user, there is no user in the context. That's why we construct a fake user to be able to call the private APIs of widgets. But passing directly this user in the API calls is not sufficient in the case the data retrieved has resolved fields because we use the context to get the current user at those places. For example the field country of administrative area.

Reply:
where exactly in the code please ?

Reply:
`administrativeArea-resolver.ts` line 21

Reply:
In public api, we call the private api, which need the `context` and, at some point, the `context.user`. As it's a public api, we don't have a user on the context, so we need to add the user retrieved from the `publicDashboard`.